### ISSUE TITLE
RESOLVED: free(): invalid pointer (segmentation fault with systemd plugin and jemalloc)

### ISSUE THREAD

**Issue by kenhys** [fluent/fluentd-kubernetes-daemonset#1517]
### Describe the bug

There is a known issue that combination with systemd plugin and jemalloc causes a
crash bug.

There are some feedback from users.

* https://github.com/fluent/fluentd-kubernetes-daemonset/issues/1484
* https://github.com/fluent/fluentd-kubernetes-daemonset/issues/1482
* https://github.com/fluent/fluentd-kubernetes-daemonset/issues/1478

Related issue:
* https://github.com/fluent/fluent-package-builder/issues/369
* https://github.com/fluent-plugins-nursery/fluent-plugin-systemd/issues/110
* https://github.com/fluent/fluentd-docker-image/issues/378

* https://github.com/fluent/fluentd-kubernetes-daemonset/issues/1406
   * #1406 may be different case.

> [!IMPORTANT]
> As a workaround, you can avoid a segmentation fault to disable jemalloc memory allocator  by specifying empty LD_PRELOAD for each container.
> ```
> env:
>   ...
>   LD_PRELOAD: ""
> ``` 

> [!CAUTION]
> UPDATE: 2024 Sep 11 
> fluent-plugin-systemd 1.1.0 (systemd-journal 2.0.0) was shipped to fix this issue.
> As v1.17.1-1.1 image bundles fluent-plugin-systemd 1.1.0 or later, please use newer image.

### To Reproduce

See above mentioned issues and related bugs.
(Especially https://github.com/fluent/fluentd-docker-image/issues/378 is the simplified to reproduce without k8s)

### Expected behavior

Even though jemalloc is enabled, it does not cause a segmentation fault)

### Your Environment

```markdown
- all of v1.16.x or later
```

### Your Configuration

```apache
See already mentioned issues.
```

### Your Error Log

```shell
See already mentioned issues.
```

### Additional context

To avoid this issue, jemalloc memory allocator will be disabled by default for v1.17.0-1.3 or later and v1.16.5-1.3 or later.

If you don't use systemd plugin at all, you can enable jemalloc memory allocator explicitly via env: parameter.

```
env:
  ...
  LD_PRELOAD="/usr/lib/libjemalloc.so.2"
```

**Comment by ashie** [comment]
https://github.com/ledbettj/systemd-journal/pull/96 will fix this issue.

**Comment by kenhys** [comment]
This issue was fixed via fluent-plugin-systemd 1.1.0 (which uses systemd-journal 2.0.0)

Now no need to disable jemalloc explicitly.

Please use v1.17.1-1.1 images which uses fluent-plugin-systemd 1.1.0 or later.

ref.
https://github.com/ledbettj/systemd-journal/pull/96
https://github.com/ledbettj/systemd-journal/pull/97
https://github.com/fluent/fluentd-docker-image/issues/378


> [!NOTE]
> Keep this issue open for a while to inform you.

**Comment by kenhys** [comment]
3 months had passed since the issue was created to raise an attention.

It's time to close it.
